src/databricks/labs/ucx/source_code/linters/files.py

```diff
@@ -260,76 +258,46 @@ class FileLinter:
-        path: Path,
+        context: LinterContext,
         inherited_tree: Tree | None = None,
-        content: str | None = None,
     ):
-        self._context = context
+        self._dependency = dependency
         self._path_lookup = path_lookup
-        self._path = path
+        self._context = context
         self._inherited_tree = inherited_tree
-        self._content = content
 
     def lint(self) -> Iterable[Advice]:
-        encoding = locale.getpreferredencoding(False)
-        try:
-            # Not using `safe_read_text` here to surface read errors
-            self._content = self._content or read_text(self._path)
-        except FileNotFoundError:
-            failure_message = f"File not found: {self._path}"
-            yield Failure("file-not-found", failure_message, 0, 0, 1, 1)
+        """Lint the file."""
+        if self._dependency.path.suffix.lower() in self._IGNORED_SUFFIXES:
             return
-        except UnicodeDecodeError:
-            failure_message = f"File without {encoding} encoding is not supported {self._path}"
-            yield Failure("unsupported-file-encoding", failure_message, 0, 0, 1, 1)
+        if self._dependency.path.name.lower() in self._IGNORED_NAMES:
             return
-        except PermissionError:
-            failure_message = f"Missing read permission for {self._path}"
-            yield Failure("file-permission", failure_message, 0, 0, 1, 1)
+        if self._dependency.path.suffix.lower() in self._NOT_YET_SUPPORTED_SUFFIXES:
+            message = f"Unsupported language for suffix: {self._dependency.path.suffix}"
+            yield Failure("unsupported-language", message, -1, -1, -1, -1)
             return
-
-        if self._is_notebook():
-            yield from self._lint_notebook()
-        else:
-            yield from self._lint_file()
-
-    def _is_notebook(self) -> bool:
-        assert self._content is not None, "Content should be read from path before calling this method"
-        # pre-check to avoid loading unsupported content
-        language = file_language(self._path)
-        if not language:
-            return False
-        return is_a_notebook(self._path, self._content)
-
-    def _lint_file(self) -> Iterable[Advice]:
-        assert self._content is not None, "Content should be read from path before calling this method"
-        language = file_language(self._path)
-        if not language:
-            suffix = self._path.suffix.lower()
-            if suffix in self._IGNORED_SUFFIXES or self._path.name.lower() in self._IGNORED_NAMES:
-                yield from []
-            elif suffix in self._NOT_YET_SUPPORTED_SUFFIXES:
-                yield Failure("unsupported-language", f"Language not supported yet for {self._path}", 0, 0, 1, 1)
-            else:
-                yield Failure("unknown-language", f"Cannot detect language for {self._path}", 0, 0, 1, 1)
+        source_container = self._dependency.load(self._path_lookup)
```

Comment:
All that logic in the code before these changes is now part of the dependency loading